### ISSUE TITLE
Unexpected error "list index out of range" for empty string

### ISSUE THREAD

**Issue by Armour** [explosion/spaCy#2554]
<img width="1325" alt="screen shot 2018-07-16 at 17 14 17" src="https://user-images.githubusercontent.com/5276065/42790082-c46e14a0-891d-11e8-8270-1b78e9c1fd0e.png">

<img width="911" alt="screen shot 2018-07-16 at 17 21 53" src="https://user-images.githubusercontent.com/5276065/42790085-c7cefa10-891d-11e8-8705-2f09ba569587.png">

Maybe should return an empty array, or instead of throwing this error from thinc.neural.op, change to display a developer-friendly message or warning for it? 

## How to reproduce the behavior

```python
import spacy
nlp = spacy.load('en')
test_empty_string = nlp('')
```

## Info about spaCy

* **spaCy version:** 2.0.11
* **Platform:** Darwin-17.7.0-x86_64-i386-64bit
* **Python version:** 3.6.6
* **Models:** en_core_web_md, en
* Installed Spacy using Conda

## Conda list

```shell
# packages in environment at /usr/local/anaconda3:
#
# Name                    Version                   Build  Channel
_ipyw_jlab_nb_ext_conf    0.1.0                    py36_0
alabaster                 0.7.11                   py36_0
anaconda                  custom           py36ha4fed55_0
anaconda-client           1.6.14                   py36_0
anaconda-navigator        1.8.7                    py36_0
anaconda-project          0.8.2                    py36_0
appnope                   0.1.0            py36hf537a9a_0
appscript                 1.0.1            py36h1de35cc_1
asn1crypto                0.24.0                   py36_0
astroid                   1.6.5                    py36_0
astropy                   3.0.3            py36h1de35cc_2
atomicwrites              1.1.5                    py36_0
attrs                     18.1.0                   py36_0
babel                     2.6.0                    py36_0
backcall                  0.1.0                    py36_0
backports                 1.0                      py36_1
backports.shutil_get_terminal_size 1.0.0                    py36_2
beautifulsoup4            4.6.0                    py36_1
bitarray                  0.8.2            py36h1de35cc_0
bkcharts                  0.2              py36h073222e_0
blas                      1.0                         mkl
blaze                     0.11.3                   py36_0
bleach                    2.1.3                    py36_0
blosc                     1.14.3               hd9629dc_0
bokeh                     0.13.0                   py36_0
boto                      2.48.0                   py36_1
bottleneck                1.2.1            py36h1d22016_1
bzip2                     1.0.6                h1de35cc_5
ca-certificates           2018.03.07                    0
certifi                   2018.4.16                py36_0
cffi                      1.11.5           py36h342bebf_0
chardet                   3.0.4                    py36_1
click                     6.7              py36hec950be_0
cloudpickle               0.5.3                    py36_0
clyent                    1.2.2                    py36_1
colorama                  0.3.9            py36hd29a30c_0
conda                     4.5.8                    py36_0
conda-build               3.10.9                   py36_0
conda-env                 2.6.0                         1
conda-verify              3.0.0                    py36_0
contextlib2               0.5.5            py36hd66e5e7_0
cryptography              2.2.2            py36h1de35cc_0
curl                      7.60.0               ha441bb4_0
cycler                    0.10.0           py36hfc81398_0
cymem                     1.31.2           py36h04f5b5a_0
cython                    0.28.3           py36h1de35cc_0
cytoolz                   0.9.0.1          py36h1de35cc_1
dask                      0.18.1                   py36_0
dask-core                 0.18.1                   py36_0
datashape                 0.5.4                    py36_1
dbus                      1.13.2               h760590f_1
decorator                 4.3.0                    py36_0
dill                      0.2.8.2                  py36_0
distributed               1.22.0                   py36_0
docutils                  0.14             py36hbfde631_0
en-core-web-md            2.0.0                     <pip>
en-core-web-sm            2.0.0                     <pip>
entrypoints               0.2.3                    py36_2
et_xmlfile                1.0.1            py36h1315bdc_0
expat                     2.2.5                hb8e80ba_0
fastcache                 1.0.2            py36h1de35cc_2
filelock                  3.0.4                    py36_0
flask                     1.0.2                    py36_1
flask-cors                3.0.6                    py36_0
freetype                  2.9.1                hb4e5f40_0
ftfy                      5.4.1                    py36_0
get_terminal_size         1.0.0                h7520d66_0
gettext                   0.19.8.1             h15daf44_3
gevent                    1.3.4            py36h1de35cc_0
glib                      2.56.1               h35bc53a_0
glob2                     0.6                      py36_0
gmp                       6.1.2                hb37e062_1
gmpy2                     2.0.8            py36h6ef4df4_2
greenlet                  0.4.13           py36h1de35cc_0
h5py                      2.8.0            py36h3c9e6ae_0
hdf5                      1.10.2               hfa1e0ec_1
heapdict                  1.0.0                    py36_2
html5lib                  1.0.1                    py36_0
icu                       58.2                 h4b95b61_1
idna                      2.7                      py36_0
imageio                   2.3.0                    py36_0
imagesize                 1.0.0                    py36_0
intel-openmp              2018.0.3                      0
ipykernel                 4.8.2                    py36_0
ipython                   6.4.0                    py36_1
ipython_genutils          0.2.0            py36h241746c_0
ipywidgets                7.2.1                    py36_0
isort                     4.3.4                    py36_0
itsdangerous              0.24                     py36_1
jbig                      2.1                  h4d881f8_0
jdcal                     1.4                      py36_0
jedi                      0.12.0                   py36_1
jinja2                    2.10                     py36_0
jpeg                      9b                   he5867d9_2
jsonschema                2.6.0            py36hb385e00_0
jupyter                   1.0.0                    py36_4
jupyter_client            5.2.3                    py36_0
jupyter_console           5.2.0                    py36_1
jupyter_core              4.4.0                    py36_0
jupyterlab                0.32.1                   py36_0
jupyterlab_launcher       0.10.5                   py36_0
keyring                   13.0.0                   py36_0
kiwisolver                1.0.1            py36h0a44026_0
lazy-object-proxy         1.3.1            py36h1de35cc_2
libcurl                   7.60.0               hf30b1f0_0
libcxx                    4.0.1                h579ed51_0
libcxxabi                 4.0.1                hebd6815_0
libedit                   3.1.20170329         hb402a30_2
libffi                    3.2.1                h475c297_4
libgfortran               3.0.1                h93005f0_2
libiconv                  1.15                 hdd342a3_7
libpng                    1.6.34               he12f830_0
libsodium                 1.0.16               h3efe00b_0
libssh2                   1.8.0                h322a93b_4
libtiff                   4.0.9                hcb84e12_1
libxml2                   2.9.8                hab757c2_1
libxslt                   1.1.32               hb819dd2_0
llvmlite                  0.24.0           py36hc454e04_0
locket                    0.2.0            py36hca03003_1
lxml                      4.2.2            py36hef8c89e_0
lzo                       2.10                 h362108e_2
markupsafe                1.0              py36h1de35cc_1
matplotlib                2.2.2            py36hbf02d85_2
mccabe                    0.6.1                    py36_1
mistune                   0.8.3            py36h1de35cc_1
mkl                       2018.0.3                      1
mkl-service               1.1.2            py36h6b9c3cc_4
mkl_fft                   1.0.2            py36h6b9c3cc_0
mkl_random                1.0.1            py36h5d10147_1
more-itertools            4.2.0                    py36_0
mpc                       1.1.0                h6ef4df4_1
mpfr                      4.0.1                h3018a27_3
mpmath                    1.0.0                    py36_2
msgpack-numpy             0.4.3                    py36_0
msgpack-python            0.5.6            py36h04f5b5a_0
multipledispatch          0.5.0                    py36_0
murmurhash                0.28.0           py36h0a44026_0
navigator-updater         0.2.1                    py36_0
nbconvert                 5.3.1                    py36_0
nbformat                  4.4.0            py36h827af21_0
ncurses                   6.1                  h0a44026_0
networkx                  2.1                      py36_0
nltk                      3.3.0                    py36_0
nose                      1.3.7                    py36_2
notebook                  5.5.0                    py36_0
numba                     0.39.0           py36h6440ff4_0
numexpr                   2.6.5            py36hf9f1f73_0
numpy                     1.14.5           py36h648b28d_4
numpy-base                1.14.5           py36ha9ae307_4
numpydoc                  0.8.0                    py36_0
odo                       0.5.1            py36hc1af34a_0
olefile                   0.45.1                   py36_0
openpyxl                  2.5.4                    py36_0
openssl                   1.0.2o               h26aff7b_0
packaging                 17.1                     py36_0
pandas                    0.23.2           py36h6440ff4_0
pandoc                    2.2.1                h1a437c5_0
pandocfilters             1.4.2                    py36_1
parso                     0.2.1                    py36_0
partd                     0.3.8            py36hf5c4cb8_0
path.py                   11.0.1                   py36_0
pathlib2                  2.3.2                    py36_0
patsy                     0.5.0                    py36_0
pcre                      8.42                 h378b8a2_0
pep8                      1.7.1                    py36_0
pexpect                   4.6.0                    py36_0
pickleshare               0.7.4            py36hf512f8e_0
pillow                    5.1.0            py36hb68e598_0
pip                       10.0.1                   py36_0
pkginfo                   1.4.2                    py36_1
plac                      0.9.6                      py_1    conda-forge
pluggy                    0.6.0                    py36_0
ply                       3.11                     py36_0
preshed                   1.0.0            py36h0a44026_0
prompt_toolkit            1.0.15           py36haeda067_0
psutil                    5.4.6            py36h1de35cc_0
ptyprocess                0.6.0                    py36_0
py                        1.5.3                    py36_0
pycodestyle               2.4.0                    py36_0
pycosat                   0.6.3            py36h1de35cc_0
pycparser                 2.18                     py36_1
pycrypto                  2.6.1            py36h1de35cc_9
pycurl                    7.43.0.2         py36hdbc3d79_0
pyflakes                  2.0.0                    py36_0
pygments                  2.2.0            py36h240cd3f_0
pylint                    1.9.2                    py36_0
pyodbc                    4.0.23           py36h0a44026_0
pyopenssl                 18.0.0                   py36_0
pyparsing                 2.2.0                    py36_1
pyqt                      5.9.2            py36h655552a_0
pysocks                   1.6.8                    py36_0
pytables                  3.4.4            py36h13cba08_0
pytest                    3.6.2                    py36_0
pytest-arraydiff          0.2              py36h39e3cac_0
pytest-astropy            0.4.0                    py36_0
pytest-doctestplus        0.1.3                    py36_0
pytest-openfiles          0.3.0                    py36_0
pytest-remotedata         0.3.0                    py36_0
python                    3.6.6                hc167b69_0
python-dateutil           2.7.3                    py36_0
python.app                2                        py36_8
pytz                      2018.5                   py36_0
pywavelets                0.5.2            py36h1d22016_2
pyyaml                    3.12             py36h1de35cc_1
pyzmq                     17.0.0           py36h1de35cc_3
qt                        5.9.6                h74ce4d9_0
qtawesome                 0.4.4            py36h468c6fb_0
qtconsole                 4.3.1            py36hd96c0ff_0
qtpy                      1.4.2                    py36_1
readline                  7.0                  hc1231fa_4
regex                     2018.06.21       py36h1de35cc_0
requests                  2.19.1                   py36_0
rope                      0.10.7                   py36_0
ruamel_yaml               0.15.42          py36h1de35cc_0
scikit-image              0.14.0           py36h0a44026_1
scikit-learn              0.19.1           py36hf9f1f73_0
scipy                     1.1.0            py36hf1f7d93_0
seaborn                   0.8.1                    py36_0
semver                    2.8.0                    py36_0
send2trash                1.5.0                    py36_0
setuptools                39.2.0                   py36_0
simplegeneric             0.8.1                    py36_2
singledispatch            3.4.0.3          py36hf20db9d_0
sip                       4.19.8           py36h0a44026_0
six                       1.11.0                   py36_1
snappy                    1.1.7                he62c110_3
snowballstemmer           1.2.1            py36h6c7b616_0
sortedcollections         1.0.1                    py36_0
sortedcontainers          2.0.4                    py36_0
spacy                     2.0.11           py36h6440ff4_2
sphinx                    1.7.5                    py36_0
sphinxcontrib             1.0                      py36_1
sphinxcontrib-websupport  1.1.0                    py36_1
sputnik                   0.9.3                    py36_0    conda-forge
spyder                    3.3.0                    py36_0
spyder-kernels            0.2.4                    py36_0
sqlalchemy                1.2.8            py36h1de35cc_0
sqlite                    3.24.0               ha441bb4_0
statsmodels               0.9.0            py36h1d22016_0
sympy                     1.1.1            py36h7f3cf04_0
tblib                     1.3.2            py36hda67792_0
termcolor                 1.1.0                    py36_1
terminado                 0.8.1                    py36_1
testpath                  0.3.1            py36h625a49b_0
thinc                     6.11.2           py36hf9f1f73_1
tk                        8.6.7                h35a86e2_3
toolz                     0.9.0                    py36_0
tornado                   5.0.2            py36h1de35cc_0
tqdm                      4.23.4                   py36_0
traitlets                 4.3.2            py36h65bd3ce_0
typing                    3.6.4                    py36_0
ujson                     1.35             py36h1de35cc_0
unicodecsv                0.14.1           py36he531d66_0
unixodbc                  2.3.6                h3efe00b_0
urllib3                   1.23                     py36_0
wcwidth                   0.1.7            py36h8c6ec74_0
webencodings              0.5.1                    py36_1
werkzeug                  0.14.1                   py36_0
wheel                     0.31.1                   py36_0
widgetsnbextension        3.2.1                    py36_0
wrapt                     1.10.11          py36h1de35cc_2
xlrd                      1.1.0                    py36_1
xlsxwriter                1.0.5                    py36_0
xlwings                   0.11.8                   py36_0
xlwt                      1.2.0            py36h5ad1178_0
xz                        5.2.4                h1de35cc_4
yaml                      0.1.7                hc338f04_2
zeromq                    4.2.5                h0a44026_0
zict                      0.1.3                    py36_0
zlib                      1.2.11               hf3cbc9b_2
```

**Comment by ines** [comment]
Thanks for the report! And this is strange – I thought we had a test for this.

I just ran the code with `v2.1.0` nightly and the new models ([see here](https://github.com/explosion/spaCy/releases/tag/v2.1.0a0) for the release) and it worked fine – so it looks like this will be resolved in the upcoming version 🙂

**Comment by carolglin-nbc** [comment]
what is the fix for this if we are still using 2.0.11?

**Comment by Armour** [comment]
I think the easy way is to filter all the empty input before calling it.
